src/devices/secplus_v2.c

```diff
@@ -0,0 +1,459 @@
+
+
+    // 280 is a cconservative guess
+    if (bitbuffer->bits_per_row[0] < 280) {
+        return DECODE_ABORT_LENGTH;
+    }
+
+    // loop though segments until we collect both parts, or run out of data
+    while (search_index < bitbuffer->bits_per_row[0]) {
+
+        search_index = bitbuffer_search(bitbuffer, 0, search_index, _preamble, _preamble_len);
+
+        if (search_index >= bitbuffer->bits_per_row[0]) {
+            break;
+        }
+
+        bitbuffer_clear(&bits);
+        next_pos = bitbuffer_manchester_decode(bitbuffer, 0, search_index + 26, &bits, 80);
+        search_index += 20;
+        if (bits.bits_per_row[0] < 42) {
+            continue; // DECODE_ABORT_LENGTH;
+        }
+
+        if (decoder->verbose) {
+            (void)fprintf(stderr, "%s: manchester_decode %d len = %u\n", __func__,
+                    0, bits.bits_per_row[0]);
+            bitrow_debugf(bits.bb[0], bits.bits_per_row[0], "%s: manchester_decoded %d", __func__, 0);
+        }
+
+        // valid = 0X00XXXX
+        // 1st 3rs and 4th bits should alway be 0
+        if (bits.bb[0][0] & 0xB0) {
+            if (decoder->verbose)
+                fprintf(stderr, "%s: DECODE_FAIL_SANITY\n", __func__);
+            continue;
+        }
+
+        // 2nd bit indicates with half of the data 
+        if (bits.bb[0][0] & 0xC0) {
+            if (decoder->verbose)
+                (void)fprintf(stderr, "%s: Set 2\n", __func__);
+            _decode_v2_half(&bits, rolling_2, &fixed_2, decoder->verbose);
+        }
+        else {
+            if (decoder->verbose)
+                (void)fprintf(stderr, "%s: Set 1\n", __func__);
+            _decode_v2_half(&bits, rolling_1, &fixed_1, decoder->verbose);
+        }
+
+        // break if we've recived both halfs
+        if (fixed_1.bits_per_row[0] > 1 && fixed_2.bits_per_row[0] > 1) {
+            break;
+        }
+
+    }
+
+    // Do was have what we need ??
+    if (fixed_1.bits_per_row[0] == 0 || fixed_2.bits_per_row[0] == 0) {
+        //  No?  Awww F'ck it then
+        if (decoder->verbose)
+            fprintf(stderr, "%s: DECODE_FAIL_SANITY\n", __func__);
+        return DECODE_FAIL_SANITY;
+    }
+
+    // Assemble rolling_1[] and rolling_2[] into rolling_digits[]
+    uint8_t rolling_digits[24] = {0};
+    uint8_t *r;
+
+    r    = rolling_digits;
+    *r++ = rolling_2[8];
+    *r++ = rolling_1[8];
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    // compute rolling_total from rolling_digits[]
+    uint32_t rolling_total = 0;
+    uint32_t rolling_temp  = 0;
+
+    for (int i = 0; i < 18; i++) {
+        rolling_temp = (rolling_temp * 3) + rolling_digits[i];
+        // fprintf(stderr, ">> %12d\t%d\n", rolling_temp, rolling_digits[i]);
+    }
+
```

Comment:
At this point, it's probably worth checking that `rolling_temp` is less than 2^28. If not, the code is invalid.

https://github.com/argilo/secplus/blob/dbe85c94372c8023c3334dcd7f112ed605c5a8de/secplus.py#L136-L137

Reply:
done